app/lib/features/chat/widgets/chats_list.dart

```diff
@@ -15,95 +15,100 @@ import 'package:logging/logging.dart';
 
 final _log = Logger('a3::chat::chats_list');
 
-class ChatsList extends ConsumerStatefulWidget {
+class ChatsList extends ConsumerWidget {
   final Function(String)? onSelected;
 
   const ChatsList({this.onSelected, super.key});
 
   @override
-  ConsumerState<ChatsList> createState() => _ChatsListConsumerState();
-}
-
-class _ChatsListConsumerState extends ConsumerState<ChatsList> {
-  @override
-  Widget build(BuildContext context) {
-    final hasSearchFilter = ref.watch(hasRoomFilters);
-    if (hasSearchFilter) {
-      return ref.watch(filteredChatsProvider).when(
-            data: (chats) {
-              if (chats.isEmpty) {
-                return SliverToBoxAdapter(
-                  child: Center(
-                    heightFactor: 10,
-                    child:
-                        Text(L10n.of(context).noChatsFoundMatchingYourFilter),
-                  ),
-                );
-              }
-              return renderList(
-                context,
-                chats.map((e) => e.getRoomIdStr()).toList(),
-              );
-            },
-            loading: () => const SliverToBoxAdapter(
-              child: Center(
-                heightFactor: 10,
-                child: CircularProgressIndicator(),
-              ),
-            ),
-            error: (e, s) {
-              _log.severe('Failed to filter convos', e, s);
-              return SliverToBoxAdapter(
-                child: Center(
-                  heightFactor: 10,
-                  child: Text(L10n.of(context).searchingFailed(e)),
-                ),
-              );
-            },
-            skipLoadingOnReload: true,
-          );
+  Widget build(BuildContext context, WidgetRef ref) {
```

Comment:
other than my comment this below is just splitting up the large build into neat separate widgets that are easy to comprehend.